### DISCUSSION TITLE
Include button that executes javascript code.

### DISCUSSION THREAD

**Discussion by jmhimara** [quarto-dev/quarto-cli#5386] [Q&A] [answered]
I'd like to include a button that executes some js code, but couldn't really figure it out. 

Here's what I tried: 

```
{ojs}
function generateRandomNumber() {
    var randomNumber = Math.floor(Math.random() * 100) + 1;
    alert("Your random number is: " + randomNumber);
  }
```
`<button onclick="generateRandomNumber()">Generate Random Number</button>`

But this doesn't seem to work. 

EDIT: Or any kind of code, it doesn't have to be JS.

**Comment by cscheid** [comment, marked as answer]
OJS is not javascript: you'll need to go through observablehq's tutorials to learn how to use their interactive features.

If you want to use plain javascript in your HTML files you can do that with raw HTML blocks:

````
```{=html}
<script>
function generateRandomNumber() {
    var randomNumber = Math.floor(Math.random() * 100) + 1;
    alert("Your random number is: " + randomNumber);
}
</script>
<button onclick="generateRandomNumber()">Generate Random Number</button>
```
````

**Reply by jmhimara** [reply]
Oh ok, thanks. 

Out of curiosity, is this possible with OJS?

**Reply by cscheid** [reply]
> Out of curiosity, is this possible with OJS?

The answer is yes, but we try not to dilute our discussion forums with questions that belong elsewhere. It's also going to be easier for you in the long run if you use resources at your disposal, like google: https://www.google.com/search?q=site%3Aobservablehq.com+button.
